### DISCUSSION TITLE
Any plans to add support for symbolic arrays?

### DISCUSSION THREAD

**Discussion by emummel20** [uiua-lang/uiua#535] [General]
- https://www.youtube.com/watch?v=U_5rU77iSPI

Curious because, coming from prolog, one thing I would love to experiment with is using [logical matrices](https://en.wikipedia.org/wiki/Logical_matrix) and [binary relations](https://en.wikipedia.org/wiki/Binary_relation), which can be handily expressed as arrays, to relate one type of array to another.

**Comment by kaikalii** [comment]
Currently, all Uiua arrays are implemented in the same way, as a data buffer with an associated shape.
While special arrays that are not represented this way are not off the table, I currently have no plans to add them.
It may be worth an experimentation though.

**Reply by emummel20** [reply]
Thanks Kai, that would be amazing. Of course no pressure whatsoever but I hope you would entertain the idea. imo there is some interesting potential there.
